test/states_and_operators.jl

```diff
@@ -189,16 +189,38 @@
         @test ψx ≈ F_3_3' * ψk
     end
 
+    @testset "random unitary" begin
+        U1 = rand_unitary(20)
+        U2 = rand_unitary(20, :exp)
+        U3 = rand_unitary([5, 5])
+        U4 = rand_unitary([5, 5], :exp)
+        @test isunitary(U1)
+        @test isunitary(U2)
+        @test isunitary(U3)
+        @test isunitary(U4)
+        @test U1.dims == U2.dims == [20]
+        @test U3.dims == U4.dims == [5, 5]
+        @inferred QuantumObject{Matrix{ComplexF64},OperatorQuantumObject} rand_unitary(10, :haar)
```

Comment:
Can you just put the type inference tests inside an inner `@testset`, and also remove the manual return type? It should be automatically detected.

See for example the other inference tests.

Reply:
Done